### DISCUSSION TITLE
Welcome to photobooth-app Discussions!

### DISCUSSION THREAD

**Discussion by mgineer85** [photobooth-app/photobooth-app#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by laguinch** [comment]
I try to update Photo Booth-app but I have this message
`pi@pibooth:~ $ pip install --upgrade photobooth-app
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.`

**Reply by laguinch** [reply]
here the solution 
step 1 : Open Terminal
step 2 : Run cd /
step 3 : Run cd etc
step 4 : Run sudo nano pip.conf
step 5 : Add following line in it after [global]
break-system-packages = true
step 6 : ctrl + x -> y -> press enter (Perform save in nano editor)
Every thing is updated now you can run pip install --upgrade photobooth-app

**Reply by mgineer85** [reply]
How did you initially install?
Using pipx?
Please try 
Pipx upgrade photobooth-App

**Reply by mgineer85** [reply]
You might end up with two installations: one install by pip and one by pipx. Either way is fine usually, but if you have two versions you might run into issues one day.

**Reply by laguinch** [reply]
i install with pipx, so I will reset all a do a clean install

**Reply by mgineer85** [reply]
I think it's fine to use pip uninstall, but did not try. 

I'll fix the update section in the docs soon to cover the different installation methods .

**Comment by samyness** [comment]
Hello @mgineer85 , 

I'm currently looking into the "Keep RAW" topic. The simplest solution would be to handle it like this:

```
# Output the original file name
original_file_name = file_path.name
logger.info(f"Original file name: {original_file_name}")

# Change file extension from .NEF to .JPG in the file name
file_path.name = original_file_name.replace(".NEF", ".JPG")
```
However, each camera brand naturally has different RAW file extensions. 
Do you have any ideas on how I could work around this?

Are the JPEGs currently being deleted anywhere? 
I couldn't find anything related to that in the script.

Thanks! 

LG Samuel

**Reply by mgineer85** [reply]
Hi Samuel,

which camera you use?
Liveview and capture normal jpg work fine?
Did you try to manually setup a gphoto command line that would do what you want? 
If it works on the commandline, we can find a way to implement in the app for sure.

**Reply by samyness** [reply]
Hey, 
I am using a Nikon D810.
Liveview and capture working great!
Adding this to the Script working fine with the first picture:
`                # capture hq picture
                logger.info("taking hq picture")
                try:
                    file_path = self._camera.capture(gp.GP_CAPTURE_IMAGE)

                    # Output the original file name
                    original_file_name = file_path.name
                    logger.info(f"Original file name: {original_file_name}")
                    logger.info(f"Original file path: {file_path.folder}/{file_path.name}")

                    # Change file extension from .NEF to .JPG in the file name
                    new_file_path_name = original_file_name.replace(".nef", ".jpg")

                    logger.info(f"New file path (JPG): {file_path.folder}/{new_file_path_name}")`

**Reply by mgineer85** [reply]
OK, so 
1) I will add the Nikon D810 to the list of working cameras to the reference :)
2) let me understand:

- you set the camera to RAW+JPEG capture?
- in that mode, the camera returns the filename of the RAW file (not the JPG, but both are captured and stored to the camera)?
- the first file download works this way.
- the second one you run into the issue described here:https://github.com/jim-easterbrook/python-gphoto2/issues/51


There seems to be a solution/workaround/explanation:
see first link, there is a discussion about dequeueing events from the camera. Even if you don't need the 2 images it seems to create 2 events and both need to be read. Second is an example linked in first issue and could help to find a solution:
- https://github.com/jim-easterbrook/python-gphoto2/issues/65#issuecomment-433615025
- https://github.com/jim-easterbrook/python-gphoto2/commit/bc686bdaf06ac764c72c3d15f700e93a2e416996

maybe you can try that?

**Reply by mgineer85** [reply]
Let us keep the discussion in the issue tracker.
Once the video feature is integrated I will think about how this could be improved.
https://github.com/photobooth-app/photobooth-app/issues/168

**Reply by samyness** [reply]
The script works with the changes provided above and `self._camera.exit()`. However, I still have the following issues: Currently, the photos are being stored in the camera's RAM, so they are not stored on the CF- Card. (I have to try this with a SD Card...) And naturally, the camera briefly loses connection after each photo, causing the mirror to fold down. However, this shouldn't be a major concern.

**Comment by samyness** [comment]
1. Yes the Nikon D810 is working great! 👍 
2.
- RAW+JPEG Capture: Yes, I have set the camera to RAW+JPEG capture mode (RAW = .nef + JPEG Fine).
- Filename Return: In this mode, the camera indeed returns the filename of the RAW file. The returned file name has a ".nef" ending. Additionally, the camera captures and stores both RAW and JPEG files, but the downloaded (RAW-)File has a .jpg ending, and therefore it is not processable.
- File Download: The first file download works successfully with the code changes. However, the second call results in a "File not found" error.
